### DISCUSSION TITLE
Main thread stack size

### DISCUSSION THREAD

**Discussion by putertubby** [mbed-ce/mbed-os#333] [Q&A] [answered]
Hello Community!

I have run in to a (likely) stack overrun issue trying to use `std::regex` in my project and would like to try increasing the size of the main thread stack to see if that changes things. I'm assuming that the stack size is 1024 bytes based on the excerpt from the linker file shown below.
~~~
    .stack_dummy (COPY):
    {
        *(.stack*)
    } > RAM
    __StackTop = ORIGIN(RAM) + LENGTH(RAM);
    _estack = __StackTop;
    __StackLimit = __StackTop - 0x400;
    PROVIDE(__stack = __StackTop);
~~~
Doing research I found [this](https://forums.mbed.com/t/mbed-ce-stm32l07x-bare-metal-stack-size/21775) thread and based on suggestions found there I changed `mbed_app.json5` to look like this.
~~~
{
    "target_overrides": {
        "*": {
            "platform.stdio-baud-rate": 115200,
            "platform.stdio-buffered-serial": 1,
            "macros_add": [
                "MBED_NO_GLOBAL_USING_DIRECTIVE"
            ]
        },
        "ARDUINO_OPTA_M7": {
            "target.boot-stack-size": "0x800"
        }
    }
}
~~~
This indeed seem to change the linker file as described in the thread but there is no change in behaviour. The thread mentions  "bare metal"  but I'm unsure about what that means and if it applies in my case? I'm also curious about if this is indeed the correct way to increase the size of the main thread stack in Mbed CE?

For context I should mention that I'm building software for an Arduino Opta leveraging the built-in boot software.

Thanks for listening!

**Comment by multiplemonomials** [comment, marked as answer]
Ah yes.  Boot stack size refers to the stack size used for the first code that boots up.  For mbed-baremetal, that's your application thread itself.  For mbed-os, that will be the RTOS kernel.  If you are using the RTOS, you should in fact be adjusting `rtos.main-thread-stack-size`, which defaults to 4096 bytes.
